package/MDAnalysis/core/universe.py

```diff
@@ -494,8 +495,22 @@ def transfer_to_memory(self, start=None, stop=None, step=None,
                 coordinates = []  # TODO: use pre-allocated array
                 for i, ts in enumerate(self.trajectory[start:stop:step]):
                     coordinates.append(np.copy(ts.positions))
+                    dimensions.append(np.copy(ts.dimensions))
                     pm.echo(i, frame=ts.frame)
                 coordinates = np.array(coordinates)
+            # If we extrated the coordinates using Timeseries, we still need
+            # to go through the trajectory to extract the dimensions.
+            else:
+                n_frames = len(range(
+                    *self.trajectory.check_slice_indices(start, stop, step)
+                ))
+                pm_format = ('{step}/{numsteps} frame dimensions '
+                             'copied to memory (frame {frame})')
+                pm = ProgressMeter(n_frames, interval=1,
+                                   verbose=verbose, format=pm_format)
+                for i, ts in enumerate(self.trajectory[start:stop:step]):
```

Comment:
That would be a good idea, indeed. Especially since, it would allow to just ditch the loop in transfer to memory.

I have one doubt, though. The way I see the API, I would add a `ProtoReader.timeseries_dimensions` method. This would mean that in all cases `transfer_to-memory` will iterate twice over the trajectory. This can be a performance issue with slow readers.

Any idea before I dive in it?